File: 128-longest-consecutive-sequence/longest-consecutive-sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
    let max = 0;
    let numsSet = new Set(nums);
    for (let n of nums) {
        console.log('n=', n);
        if (!numsSet.has(n - 1)) {
            let count = 1;
            while (numsSet.has(n + 1)) {
                count++;
                n++;
            }
            max = Math.max(max, count);
        }
    }
    return max;
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums_set = set(nums)
        best = 0
        for n in nums:
            if (n - 1) not in nums_set:
                count = 1
                while (n + 1) in nums_set:
                    count += 1
                    n += 1
                best = max(count, best)
        return best